pulp_ansible/app/serializers.py

```diff
@@ -648,11 +811,51 @@ class CollectionVersionSerializer(ContentChecksumSerializer, CollectionVersionUp
 
     creating = True
 
+    def create(self, validated_data):
+        artifact = validated_data.get("artifact")
+        repository = validated_data.get("repository")
+        sigstore_verifying_services = repository.sigstore_verifying_service.all()
+
+        cv_instance = super().create(validated_data)
+
+        artifact_name = artifact.file.name
+        artifact_file = storage.open(artifact_name)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            with tarfile.open(fileobj=artifact_file, mode="r") as tar:
+                tar.extractall(path=tempdir)
+
+                for root, dirs, files in os.walk(tempdir):
+                    if ".ansible-sign" in dirs:
+                        manifest_path = Path(tempdir) / ".ansible-sign" / "sha256sum.txt"
+                        signature_path = Path(tempdir) / ".ansible-sign" / "sha256sum.txt.sigstore"
```

Comment:
I take it this means that collection tarballs can have signatures inside now? Does this affect the digest of the collection's manifest file?

Reply:
Yes, `ansible-sign` will generate signatures under the mentioned path, which should be extracted and verified when a collection version gets uploaded to a repository. As far as I understand, this should not be computed as part of the manifest file so it shouldn't affect its digest.